### PR TITLE
Move the lock to the resource-group.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/ble_esp32.cc [hunk 1]
```diff
@@ -227,12 +221,26 @@ bool TokenResourceMap::resize(int new_capacity) {
...
+
+  ~BleResourceGroup() override {
+    OS::dispose(mutex_);
+  }
 
   uint32_t on_event(Resource* resource, word data, uint32_t state) override;
 
   TokenResourceMap token_resource_map;
+
+  Mutex* mutex() {
```
Comment (kasperl) [inline src/resources/ble_esp32.cc:236]:
```suggestion
  Mutex* mutex() const {
```

File: src/resources/ble_esp32.cc [hunk 2]
```diff
@@ -227,12 +221,26 @@ bool TokenResourceMap::resize(int new_capacity) {
...
+    return mutex_;
+  }
+
+ private:
+  /// A mutex protecting BLE operations.
+  /// NimBLE has its own thread, and we use this mutex to coordinate operations.
+  Mutex* mutex_ = null;
```
Comment (kasperl) [inline src/resources/ble_esp32.cc:243]: Drop the initialization.

```suggestion
  Mutex* mutex_;
```

File: src/resources/ble_esp32.cc [hunk 3]
```diff
@@ -1265,6 +1271,9 @@ static Object* convert_mbuf_to_heap_object(Process* process, const os_mbuf* mbuf
   return data;
 }
 
+BleCallbackScope::BleCallbackScope()
+      : locker(BleAdapterResource::instance()->group()->mutex()) {}
```
Comment (kasperl) [inline src/resources/ble_esp32.cc:1275]:
```suggestion
    : locker(BleAdapterResource::instance()->group()->mutex()) {}
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/resources/ble_esp32.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/ble_esp32.cc]: done.

File: src/resources/ble_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/ble_esp32.cc]: done.

File: src/resources/ble_esp32.cc (hunk 3 above)
Comment (floitsch) [reply src/resources/ble_esp32.cc]: done.

**Pull request merged**